src/components/NucleotideEntropy.tsx

```diff
@@ -0,0 +1,706 @@
+  control: (styles: CSSObjectWithLabel) => ({ ...styles, backgroundColor: 'white' }),
+  multiValue: (styles: CSSObjectWithLabel, { data }: { data: GeneOption }) => {
+    const color = chroma(data.color);
+    return {
+      ...styles,
+      backgroundColor: color.alpha(0.1).css(),
+    };
+  },
+  multiValueLabel: (styles: CSSObjectWithLabel, { data }: { data: GeneOption }) => ({
+    ...styles,
+    color: data.color,
+  }),
+  multiValueRemove: (styles: CSSObjectWithLabel, { data }: { data: GeneOption }) => {
+    return {
+          ...styles,
+          'color': data.color,
+          ':hover': {
+            backgroundColor: data.color,
+            color: 'white',
+            cursor: 'pointer',
+          },
+        };
+  },
+};
+
+function assignColorsToGeneOptions<T extends { label: string }>(options: T[]): (T & { color: string })[] {
+  const colors = mapLabelsToColors(options.map(o => o.label));
+  return options.map((o, i) => ({
+    ...o,
+    color: colors[i],
+  }));
+}
+
+const genes = jsonRefData.genes;
+!genes.map(o => o.name).includes('All') &&
+  genes.push({ name: 'All', startPosition: 0, endPosition: 29903, aaSeq: ''});
+
+const options: GeneOption[] = assignColorsToGeneOptions(genes.map(g => {
+  return {
+    value: g.name,
+    label: g.name,
+    startPosition: g.startPosition,
+    endPosition: g.endPosition,
+    aaSeq: g.aaSeq,
+    color: chroma.random().darken().hex()
+  };
+}));
+
+
+export const NucleotideEntropy = ({ selector }: Props) => {
+  const [plotType, setPlotType] = useState<string>('time');
+  const [deletions, setDeletions] = useState<boolean>(false);
+  const [empty, setEmpty] = useState<boolean>(false);
+  const [sequenceType, setSequenceType] = useState<SequenceType>('nuc');
+  const [threshold, setThreshold] = useState(0.00001);
+  const [gene, setGene] = useState<string>('all');
+  const [selectedGeneOptions, setSelectedGenes] = useState([
+    {
+      value: 'All',
+      label: 'All',
+      color: '#353B89'
+    },
+  ]);
+
+  const handleDeletions = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setDeletions(event.target.checked);
+  };
+
+  const handleEmpty = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setEmpty(event.target.checked);
+  };
+  
+  let selectedGenes = options.filter(g => selectedGeneOptions.map(o => o.value).includes(g.value));
+
+  const data = useData(selector, sequenceType, selectedGenes, deletions, empty);
+
+  const onChange = (value: any, { action, removedValue }: any) => {
+    switch (action) {
+      case 'remove-value':
+      case 'pop-value':
+        if (removedValue.isFixed) {
+          return;
+        }
+        break;
+      case 'clear':
+        value = [{ value: 'All', label: 'All', color: '#353B89' }];
+        break;
+    }
+    setSelectedGenes(value);
+  };
+
+  const controls = (
+    <div className='mb-4'>
+      {/* AA vs. nucs */}
+      <div className='mb-2 flex'>
```

Comment:
I would guess that we could extract some of those components to reuse them here and in `NucleotideEntropyMultiChart`